presto-native-execution/presto_cpp/main/PeriodicMemoryChecker.cpp

```diff
@@ -211,4 +214,22 @@ void PeriodicMemoryChecker::pushbackMemory() {
       kCounterMemoryPushbackLatencyMs, latencyMs * 1000);
   LOG(INFO) << "Shrunk " << velox::succinctBytes(freedBytes);
 }
+
+void startPeriodicMemoryChecker() {
+#ifdef __linux__
+  PeriodicMemoryChecker::Config config;
+  auto* systemConfig = SystemConfig::instance();
+  config.systemMemPushbackEnabled = systemConfig->systemMemPushbackEnabled();
+  config.systemMemLimitBytes = systemConfig->systemMemLimitGb() << 30;
+  config.systemMemShrinkBytes = systemConfig->systemMemShrinkGb() << 30;
+  createLinuxMemoryChecker(config);
+  linuxMemoryChecker()->start();
```

Comment:
This is then changed to get the singleton instance and checks if the checker was instantiated. If it was, then it can be started/stopped.